cgi-bin/DW/API/Key.pm

```diff
@@ -57,12 +57,17 @@ sub new_for_user {
 sub get_key {
     my ( $class, $hash ) = @_;
     return undef unless $hash;
-
-    my $dbh     = LJ::get_db_writer() or croak "Failed to get database";
-    my $keydata = $dbh->selectrow_hashref(
-        "SELECT keyid, userid, hash FROM api_key WHERE hash = ? AND state = 'A'",
-        undef, $hash );
-    carp $dbh->errstr if $dbh->err;
+    my $memkey = [ $hash, "api_key:" . $hash ];
```

Comment:
I would recommend just using

```perl
my $memkey = "api_key:$hash";
```

The reason you would use the `[ X, Y ]` format is only if you are going to be getting/setting multiple related Y's and you want them to map to the same underlying Memcache server.

For example, we frequently use something like `[ $userid, "thingy:$userid" ]` because that lets us put all of that user's data onto the same Memcache server. This makes it more efficient to do a multi-get or multi-set, also means that if you have a memcache outage, it affects 100% of data for X% of users rather than X% of all data for 100% of users.